compiler/src/iree/compiler/Dialect/Flow/Transforms/RaiseSpecialOps.cpp

```diff
@@ -245,42 +262,50 @@ tryRaiseToExtractSlice(AffineMap inputIndexingMap, AffineMap outputIndexingMap,
   IntegerAttr zero = rewriter.getI64IntegerAttr(0);
   IntegerAttr one = rewriter.getI64IntegerAttr(1);
   unsigned currOutDim = 0;
+  unsigned leadOutDim = 0;
   for (auto [idx, expr] : llvm::enumerate(inputIndexingMap.getResults())) {
-    // Check if the input dimension matches the current output dimension.
-    if (expr == outputIndexingMap.getResult(currOutDim)) {
-      offsets.push_back(zero);
-      // Get the dim size from the output tensor.
-      if (outShape[currOutDim] == ShapedType::kDynamic) {
-        auto dim = rewriter.create<tensor::DimOp>(linalgOp.getLoc(), output,
-                                                  currOutDim);
-        sizes.push_back(dim.getResult());
-      } else {
-        sizes.push_back(rewriter.getI64IntegerAttr(outShape[currOutDim]));
-      }
-      ++currOutDim;
-      continue;
-    }
-    // Assume that the constant access is a rank reducing access.
+    // Constant accesses can either be rank reducing or an access into a unit
```

Comment:
This is actually limiting the cases covered? The constant access does not have to be for dimensions that are unit dim. You could have a constant access where the source of the unit dim could be anything. Then the constant (or any ssa value for that matter that is not related the Linalg op loops become the offset, and the size is 1.

I'd like to take a bit of a further look into this.

Reply:
I think the comment here is a bit misleading. It does have to either be a unit dim or rank reducing on the output though due to the constraints of a linalg op. If not one of those then it is a broadcast and thus can't be raised, but I think that should be captured properly. That said, I might not have thought through it properly and will come back to it later. I'm also starting to think that this intermediate linalg approach is nice for the cases that work, but there are certain classes of slices (exactly when we have an arbitrary computation of an index/size for slicing) that warrant a direct lowering. I'd like to see real world use cases for those before adding raisings for them though.

I'll add a test case for broadcast to make sure it isn't erroneously raised though.